controllers/sessionController.js

```diff
@@ -0,0 +1,51 @@
+const User = require("../models/User");
+const parseVErr = require("../utils/parseValidationErrs");
+const csrf = require("host-csrf");
+
+const registerShow = (req, res) => {
+  res.render("register");
+};
+
+const registerDo = async (req, res, next) => {
+  if (req.body.password != req.body.password1) {
+    req.flash("error", "The passwords entered do not match.");
+    res.render("register");
+  }
+  try {
+    await User.create(req.body);
+  } catch (e) {
+    if (e.constructor.name === "ValidationError") {
+      parseVErr(e, req);
+    } else if (e.name === "MongoServerError" && e.code === 11000) {
+      req.flash("error", "That email address is already registered.");
+    } else {
+      return next(e);
+    }
+    return res.render("register");
```

Comment:
Here, you also must to pass the error as the second parameter for the render method.